File: S2M1/src/FactoriaCarreraMontaña.java
public class FactoriaCarreraMontaña implements FactoriaCarreraYBicicleta {

	public Carrera crearCarrera() {
		return new CarreraMontaña();
	}

	public Bicicleta crearBicicleta(int dorsal, boolean tengoAcabar) {
		return new BicicletaMontaña(dorsal, tengoAcabar);
	}
}

File: S2M1/Ruby/FactoriaCarreraMontaña.rb
class FactoriaCarreraMontaña < FactoriaCarreraYBicicleta

	def crearCarrera()
		return CarreraMontaña.new()
    end

	def crearBicicleta(dorsal, tengoAcabar) 
		return BicicletaMontaña.new(dorsal, tengoAcabar)
    end
end